python/tvm/target.py

```diff
@@ -134,29 +134,66 @@ def __repr__(self):
         return self.__str__()
 
     def __enter__(self):
-        self._old_target = Target.current
-        if self._old_target is not None and str(self) != str(self._old_target):
-            warnings.warn(
-                "Override target '%s' with new target scope '%s'" % (
-                    self._old_target, self))
-        Target.current = self
+        _api_internal._EnterTargetScope(str(self))
         return self
 
     def __exit__(self, ptype, value, trace):
-        Target.current = self._old_target
+        _api_internal._ExitTargetScope()
 
+@register_node
+class GenericFunc(NodeBase):
+    """GenericFunc node reference, for internal use in generic_func.
 
-def generic_func(fdefault):
+    Note
+    ----
+    Do not construct an instance of this object, it should only ever be
+    used as a return value from calling into C++.
+    """
+    def __call__(self, *args):
+        return _api_internal._GenericFuncCallFunc(self, *args)
+
+    def set_default(self, func, allow_override):
+        _api_internal._GenericFuncSetDefault(self, func, allow_override)
+
+    def register_func(self, func, key_list, allow_override):
```

Comment:
make API consistent with old generic_func both in c++ and python

- register
- set_default